hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/record/SingleTransactionRecordBuilder.java

```diff
@@ -31,4 +31,15 @@ public interface SingleTransactionRecordBuilder {
      */
     @NonNull
     ResponseCodeEnum status();
+
+    /**
+     * Sets the status for this Record Builder.
+     * Most common use for this is when a child transaction fails and the parent record must be
+     * set to fail, or a parent transaction determines the child status must be modified due to
+     * subsequent processing.
+     * @param status a {@link ResponseCodeEnum} representing the new status to assign for this record.
+     * @return this record builder instance.
+     */
+    @NonNull
+    public SingleTransactionRecordBuilder status(@NonNull final ResponseCodeEnum status);
```

Comment:
A handler should throw a `HandleException` to signal failures. The workflow will ensure that all changes are rolled back and the status of the record and all child records is updated.

Reply:
These aren't failures.  That's my point.  There are multiple non-failure result codes, and current behavior uses them all.  How do I match that when I cannot set (e.g.) SUCCESS rather than OK?

Reply:
OK is the one I've seen in use, not sure if there are others.